depthai_nodes/parser_generator.py

```diff
@@ -53,7 +54,22 @@ def build(self, nn_archive: dai.NNArchive, head_index: int = None) -> Dict:
                 )
 
             head = decode_head(head)
-            parsers[index] = pipeline.create(parser).build(head)
+            sig = inspect.signature(parser.build)
+            if "inputs_size" in sig.parameters:
+                inputs_size = []
+                for input in nn_archive.getConfig().model.inputs:
+                    if input.layout == "NHWC":
+                        _, height, width, _ = input.shape
+                    elif input.layout == "NCHW":
+                        _, _, height, width = input.shape
+                    else:
+                        raise ValueError(
+                            f"Input layout {input.layout} not supported for input_size extraction."
+                        )
+                    inputs_size.append([width, height])
```

Comment:
This should be List of Tuples
```suggestion
                    inputs_size.append(tuple([width, height]))
```

Reply:
The `inputs_size` parameter is currently defined as `List[List[int]]`. So this would also require a change there. Do we prefer a list of Tuples instead of a list of Lists?